contracts/contracts/subnet/SubnetActorManagerFacet.sol

```diff
@@ -177,31 +177,56 @@ contract SubnetActorManagerFacet is SubnetActorModifiers, ReentrancyGuard, Pausa
         }
     }
 
-    function _patchGenesisValidators(address validator) internal {
+    function _increaseGenesisValidatorCollateral(address validator, uint256 amount) internal {
         // add to initial validators avoiding duplicates if it
         // is a genesis validator.
         bool alreadyValidator;
         uint256 length = s.genesisValidators.length;
-        for (uint256 i; i < length; ) {
+        uint256 i;
+        for (; i < length; ) {
             if (s.genesisValidators[i].addr == validator) {
-                alreadyValidator = true;
                 break;
             }
             unchecked {
                 ++i;
             }
         }
-        if (!alreadyValidator) {
+        // already validator if we iterate through all of them but didn't get a match
+        if (i >= length) {
+            s.genesisValidators[i].weight += amount;
+        } else {
             uint256 collateral = s.validatorSet.validators[validator].currentPower;
```

Comment:
do we still need this line?

Reply:
Hmm, I don't know. I think this is the piece of the original code I did not fully grasp. Why are we setting the new power to the `current` power, rather than the actual amount, since it didn't exist before?